docker/scripts/entrypoint-replication.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+###############################################################################
+# This a wrapper script for ppdb-replication which is intended to run in a
+# Docker container. Configuration is managed by environment variables, which
+# are defined in the Phalanx application, so that no command line arguments are
+# required.
+###############################################################################
+
+# Bash "strict mode", to help catch problems and bugs in the shell script.
+# Every bash script you write should include this. See
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/ for
+# details.
+set -euo pipefail
+
+# Check if the command is found
+command -v ppdb-replication >/dev/null 2>&1 || { echo "ppdb-replication command not found"; exit 1; }
+echo "Found ppdb-replication command"
+
+# Function to check if an environment variable is set
+check_env_var() {
+    local var_name="$1"
+    local var_value="${!var_name}"
+    if [ -z "${var_value:-}" ]; then
+        echo "$var_name is a required environment variable"
+        exit 1
+    fi
+}
+
+check_env_var "PPDB_REPLICATION_APDB_CONFIG"
+check_env_var "PPDB_REPLICATION_PPDB_CONFIG"
+
+# Build the command from the environment variables
+_CMD="ppdb-replication"
+[ -n "${PPDB_REPLICATION_MON_LOGGER:-}" ] && _CMD="$_CMD --mon-logger $PPDB_REPLICATION_MON_LOGGER"
+[ -n "${PPDB_REPLICATION_MON_RULES:-}" ] && _CMD="$_CMD --mon-rules $PPDB_REPLICATION_MON_RULES"
+[ -n "${PPDB_REPLICATION_LOG_LEVEL:-}" ] && _CMD="$_CMD -l $PPDB_REPLICATION_LOG_LEVEL"
+_CMD="$_CMD run"
+[ "${PPDB_REPLICATION_UPDATE_EXISTING:-}" = "true" ] && _CMD="$_CMD --update"
+[ -n "${PPDB_REPLICATION_MIN_WAIT_TIME:-}" ] && _CMD="$_CMD --min-wait-time $PPDB_REPLICATION_MIN_WAIT_TIME"
+[ -n "${PPDB_REPLICATION_MAX_WAIT_TIME:-}" ] && _CMD="$_CMD --max-wait-time $PPDB_REPLICATION_MAX_WAIT_TIME"
+[ -n "${PPDB_REPLICATION_CHECK_INTERVAL:-}" ] && _CMD="$_CMD --check-interval $PPDB_REPLICATION_CHECK_INTERVAL"
+[ -n "${PPDB_REPLICATION_APDB_CONFIG:-}" ] && _CMD="$_CMD $PPDB_REPLICATION_APDB_CONFIG"
+[ -n "${PPDB_REPLICATION_PPDB_CONFIG:-}" ] && _CMD="$_CMD $PPDB_REPLICATION_PPDB_CONFIG"
+
+# Run the command
+echo "Running: $_CMD"
+$_CMD
+
+# Check the exit status of the command
+status=$?
+if [ $status -ne 0 ]; then
+    echo "Command failed with exit status $status"
+    exit $status
+fi
```

Comment:
With `set -e` if `$_CMD` fails then the shell will exit immediately, this `if` is only ever executed if status=0.

Reply:
Good point, will remove it.